cassiopeia/simulator/Cas9LineageTracingDataSimulator.py

```diff
@@ -0,0 +1,328 @@
+
+    This subclass of `LineageTracingDataSimulator` extends the `overlay_data`
+    function to simulate Cas9-based mutations onto a defined "cassette". These
+    cassettes emulate the TargetSites described in Chan et al, Nature 2019 and
+    the GESTALT arrays described in McKenna et al, Science 2020 in which several
+    Cas9 cut-sites are arrayed together. In the Chan et al technology, these
+    cassettes are of length 3; in McKenna et al, these cassettes are of length
+    10.
+
+    The class accepts several parameters that govern the Cas9-based tracer.
+    First and foremost is the cutting rate, describing how fast Cas9 is able to
+    cut. We model Cas9 cutting as an exponential process, parameterized by the
+    specified mutation rate - specifically, for a lifetime `t` and parameter
+    `lambda`, the expected probability of Cas9 mutation, per site, is
+    `exp(-lambda * t)`.
+    
+    Second, the class accepts architecture of the recorder - described by the
+    size of the cassette (by default 3) and the number of cassettes. The
+    resulting lineage will have |cassette|*(# of cassettes) characters.
+
+    Third, the class accepts a state distribution describing the relative
+    likelihoods of various indels. This is very useful, as it is typical that a
+    handful of mutations are far likelier than the bulk of the possible
+    mutations.
+
+    Finally, the class accepts two types of silencing rates. The first is the
+    heritable silencing rate which is a rare event in which an entire cassette
+    is transcriptionally silenced and therefore not observed. The second type
+    of silencing is a stochastic dropout rate which simulates the loss of
+    cassettes due to the low sensitivity of the RNA-sequencing assay. 
+
+    The function `overlay_data` will operate on the tree in place and will
+    specifically modify the data stored in the character attributes.
+
+    Args:
+        number_of_cassettes: Number of cassettes (i.e., arrays of target sites)
+        size_of_cassette: Number of editable target sites per cassette
+        mutation_rate: Exponential parameter for the Cas9 cutting rate.
+        state_generating_distribution: Distribution from which to simulate state
+            likelihoods. This is only used if mutation priors are not
+            specified to the simulator.
+        number_of_states: Number of states to simulate
+        mutation_priors: A mapping from state to probability that a user
+            can specify. If this argument is not None, states will not be
+            pulled from the state distribution.
+        heritable_silencing_rate: Silencing rate for the cassettes, per node,
+            simulating heritable missing data events.
+        stochastic_silencing_rate: Rate at which to randomly drop out cassettes,
+            to simulate dropout due to low sensitivity of assays.
+        random_seed: Random seed to use for deterministic simulations. Note
```

Comment:
I think we should be more precise and say that this sets the *numpy* random seed.